exploration/0477-quoted-literals.md

```diff
@@ -0,0 +1,192 @@
+# Quoted Literals
+
+Status: **Accepted**
+
+<details>
+	<summary>Metadata</summary>
+	<dl>
+		<dt>Pull Request</dt>
+		<dd><a href="https://github.com/unicode-org/message-format-wg/pull/477">#477</a></dd>
+	</dl>
+</details>
+
+## Objective
+
+Document the rationale for including quoted literals in MessageFormat and for delimiting them with the vertical line character, `|`.
+
+## Background
+
+MessageFormat allows both quoted and unquoted literals. Unquoted literals satisfy many common use-cases for literals: they are sufficient to represent numbers and single-word option values and variant keys. Quoted literals are helpful in exotic use-cases.
+
+In early drafts of the MessageFormat syntax, quoted literals used to be delimited first with quotation marks (`"foo bar"`), and then with round parentheses, e.g. `(foo bar)`. See [#263](https://github.com/unicode-org/message-format-wg/issues/263).
+
+In [#414](https://github.com/unicode-org/message-format-wg/pull/414) proposed to revert these changes and go back to using single and/or double quotes as delimiters. The propsal was rejected. This document is an artifact of that rejection.
+
+## Use-Cases
+
+_What use-cases do we see? Ideally, quote concrete examples._
+
+In general, quoted literals are useful for:
+
+1. encoding literals containing whitespace, like literals consisting of multiple words,
+1. encoding literals containing exotic characters that do not conform to the `unquoted` production in ABNF.
+
+More specifically:
+
+- Message authors and translators need to be able to use the apostrophe in the message content, and may want to use the single quote character to represent it instead of the typograhic (curly) apostrophe.
+
+  > ```
+  > {…{|New Year's Eve|}…}
+  > ```
```

Comment:
This is the wrong example. The problem isn't the occasional use of apostrophes in literals but the unavailability of apostrophe (and double-quote) in the message body:

```suggestion
>```
>{You can't have more than {$num :number} {'bone dragons'}.}
>```
```
```

Reply:
It’s legal to have quotes inside patterns. That’s by design, and the consequence of our using curly braces as pattern delimiters. This use-case was meant to be about the occasional use of apostrophe inside literals :)